README.md

```diff
@@ -25,3 +25,11 @@ for you to talk with someone who **does** know the code, or who wants
 to pair with you and figure it out.  Then you can work on writing up
 what you learned.
 
+To help prevent accidentally introducing broken links, we use the
+`mdbook-linkcheck`. If installed on your machine `mdbook` will automatically
+invoke this link checker, otherwise it will emit a warning saying it couldn't
+be found.
```

Comment:
what kind of links does this check? can it test `https` links too?

Reply:
It reads the contents of the raw chapters then uses `pulldown-cmark` to find all the links on a page. Then if it's a link to another chapter we make sure that chapter exists, otherwise if its http(s) it'll try to `GET` the page using `reqwests`. 

Fetching pages from the web is disabled by default because it can take a while when your book has lots of them. Therefore you need to set `output.linkcheck.follow-web-links` to `true` in your `book.toml`.